### PR TITLE
Destination Snowflake Cortex: SQL Processor Refactor

### PR DESCRIPTION
**Pull request by aaronsteers**
## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/catalog/catalog_providers.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/catalog/catalog_providers.py:107]: This is the only new method on this class. The rest is copied directly from PyAirbyte.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/destinations/record_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/destinations/record_processor.py:1]: Nothing changed here. Copied directly from PyAirbyte.

File: airbyte-cdk/python/pyproject.toml
Comment (aaronsteers) [inline airbyte-cdk/python/pyproject.toml:122]: Lint and format settings copied from PyAirbyte.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/state/state_writers.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/state/state_writers.py:1]: Nothing changed here. Copied from PyAirbyte.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/globals.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/globals.py:11]: Moved to a central 'globals' file so multiple classes can depend on these without risking circular dependencies.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/indexer.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/indexer.py:1]: The 'Indexer' class is fully replaced by the `SnowflakeCortexSqlProcessor` class

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/destination.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/destination.py:42]: Indexer is replaced by SqlProcessor, and Embedding functions are moved into the SqlProcessor so that they can be performed more efficiently: in bulk, rather than on a per-record/per-document basis. This also means that we never have to create multiple record objects for the same source records, and we don't have to create an adapted catalog. The source catalog is just the source catalog, and the storage of the record into different columns is handled internally by the SqlProcessor.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/destination.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/destination.py:75]: I'll remove this.

- [ ] TODO

File: airbyte-integrations/connectors/destination-snowflake-cortex/unit_tests/indexer_test.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/unit_tests/indexer_test.py:1]: Since there is no more Indexer class, this test file is deleted. In a future pass, I'd like to create more unit tests which replicate/replace these tests on the now-removed Indexer class. I attempted to adapt it, but it turned into more time than it was worth at this stage.

File: pyproject.toml
Comment (aaronsteers) [inline pyproject.toml:35]:
```suggestion
```

File: pyproject.toml
Comment (aaronsteers) [inline pyproject.toml:88]:
```suggestion
```

File: airbyte-cdk/python/pyproject.toml
Comment (aaronsteers) [inline airbyte-cdk/python/pyproject.toml:1]: - [ ] TODO: Revert changes to this CDK file. This is left-over from the original plan of updating the CDK.

File: airbyte-cdk/python/poetry.lock
Comment (aaronsteers) [inline airbyte-cdk/python/poetry.lock:1]: - [ ] TODO: Revert changes to this CDK file. This is left-over from the original plan of updating the CDK.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/sql/sql_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/sql/sql_processor.py:726]: This is a new addition to the `SqlProcessor` logic. Now, `WriteStrategy.AUTO` will defer to the configured catalog's "destination sync mode".

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/sql/sql_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/sql/sql_processor.py:1]: This file is basically copied directly from PyAirbyte. The only change is noted in a comment, below, where we add handling to detect DestinationSyncMode from a configured catalog.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py:171]: This part is new. Returns a static column list regardless of source schema.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py:214]: This is untested, but the idea would be to inject a SQL-based embedding here if native embedding is requested.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py:327]: Rather than write the raw source record to JSONL, we write the chunked and embedded record to JSONL, and all records will later be loaded in bulk from those files into the Snowflake destination table.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py:316]: TODO:

- [ ] Consider changing this to int or ULID.

File: airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py:364]: TODO:

- [ ] Consider refactoring to leverage `_get_sql_column_definitions()`

File: docs/integrations/destinations/snowflake-cortex.md
Comment (aaronsteers) [inline docs/integrations/destinations/snowflake-cortex.md:10]:
```suggestion
* Snowflake Connection - where to store the vectors. This configures a vector store using Snowflake tables having the `VECTOR` data type.
```

**Review by bindipankhudi — COMMENTED** [review]
Took another pass. Looks great! 👍 We should make sure all the tests run before we merge.

**Review by bindipankhudi — APPROVED** [review]

**Pull request merged**
